modules/cfe_testcase/src/resource_id_misc_test.c

```diff
@@ -0,0 +1,163 @@
+/*************************************************************************
+**
+**      GSC-18128-1, "Core Flight Executive Version 6.7"
+**
+**      Copyright (c) 2006-2019 United States Government as represented by
+**      the Administrator of the National Aeronautics and Space Administration.
+**      All Rights Reserved.
+**
+**      Licensed under the Apache License, Version 2.0 (the "License");
+**      you may not use this file except in compliance with the License.
+**      You may obtain a copy of the License at
+**
+**        http://www.apache.org/licenses/LICENSE-2.0
+**
+**      Unless required by applicable law or agreed to in writing, software
+**      distributed under the License is distributed on an "AS IS" BASIS,
+**      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+**      See the License for the specific language governing permissions and
+**      limitations under the License.
+**
+** File: resource_id_misc_test.c
+**
+** Purpose:
+**   Functional test of Miscellaneous Resource Id APIs
+**
+**   Demonstration of how to register and use the UT assert functions.
+**
+*************************************************************************/
+
+/*
+ * Includes
+ */
+
+#include "cfe_test.h"
+/* Needed for CFE_RESOURCEID_MAKE_BASE macro */
+#include "cfe_resourceid_basevalue.h"
+
+bool TestReturnFalse(CFE_ResourceId_t CheckId)
+{
+    return false;
+}
+
+bool TestReturnTrue(CFE_ResourceId_t CheckId)
+{
+    return true;
+}
+
+void TestToFromInteger(void)
+{
+    UtPrintf("Testing: CFE_ResourceId_ToInteger, CFE_ResourceId_FromInteger, CFE_ResourceId_Equal");
+    uint32           Id1            = 1;
+    uint32           Id2            = 999;
+    CFE_ResourceId_t ResourceId1    = CFE_ResourceId_FromInteger(Id1);
+    CFE_ResourceId_t ResourceId2    = CFE_ResourceId_FromInteger(Id2);
+    CFE_ResourceId_t ResourceId1Exp = Id1;
+    CFE_ResourceId_t ResourceId2Exp = Id2;
```

Comment:
For example, right here "Id1" cannot be directly assigned to a ResourceId value.  The fact that it is permitted by default is for backward compatibility with old code.  New code should always use the proper conversion to make it explicit/clear when you really want to interpret as a plain number (In this case the `CFE_ResourceId_FromInteger` just like above).

Reply:
done :+1: